security/sshdo/files/macports.patch

```diff
@@ -0,0 +1,38 @@
+--- Makefile.orig	2023-06-19 15:23:20.000000000 +1000
++++ Makefile	2023-06-19 15:58:11.000000000 +1000
+@@ -58,8 +58,8 @@
+ 	@echo "make diff      - Show differences between source and installed versions"
+ 
+ test:
+-	@[ -z "`which python3 2>/dev/null | grep '^/'`" -a -n "`which python 2>/dev/null | grep '^/'`" ] && sed 's/env python3$$/env python/' < ./sshdo > ./sshdo2 && chmod 755 ./sshdo2 && mv ./sshdo2 ./sshdo || true
+-	@[ -z "`which python3 2>/dev/null | grep '^/'`" -a -n "`which python 2>/dev/null | grep '^/'`" ] && sed 's/env python3$$/env python/' < ./test_sshdo > ./test_sshdo2 && chmod 755 ./test_sshdo2 && mv ./test_sshdo2 ./test_sshdo || true
++	@[ -z "`which python@PYTHON_BRANCH@ 2>/dev/null | grep '^/'`" -a -n "`which python 2>/dev/null | grep '^/'`" ] && sed 's/env python3\.11$$/env python/' < ./sshdo > ./sshdo2 && chmod 755 ./sshdo2 && mv ./sshdo2 ./sshdo || true
++	@[ -z "`which python@PYTHON_BRANCH@ 2>/dev/null | grep '^/'`" -a -n "`which python 2>/dev/null | grep '^/'`" ] && sed 's/env python3\.11$$/env python/' < ./test_sshdo > ./test_sshdo2 && chmod 755 ./test_sshdo2 && mv ./test_sshdo2 ./test_sshdo || true
+ 	@./test_sshdo
+ 
+ check: test
+@@ -69,7 +69,7 @@
+ 	if [ ! -f $(DEST_ETCDIR)/$(SSHDOERS) ]; then cp $(SSHDOERS) $(DEST_ETCDIR); chmod 644 $(DEST_ETCDIR)/$(SSHDOERS); fi
+ 	if [ ! -f $(DEST_ETCDIR)/$(SSHDO_BANNER) ]; then cp $(SSHDO_BANNER) $(DEST_ETCDIR); chmod 644 $(DEST_ETCDIR)/$(SSHDO_BANNER); fi
+ 	cp $(BIN) $(DEST_BINDIR); chmod 755 $(DEST_BINDIR)/$(BIN)
+-	[ -z "`which python3 2>/dev/null | grep '^/'`" -a -n "`which python 2>/dev/null | grep '^/'`" ] && sed 's/env python3$$/env python/' < $(BIN) > $(DEST_BINDIR)/$(BIN) || true
++	[ -z "`which python@PYTHON_BRANCH@ 2>/dev/null | grep '^/'`" -a -n "`which python 2>/dev/null | grep '^/'`" ] && sed 's/env python3\.11$$/env python/' < $(BIN) > $(DEST_BINDIR)/$(BIN) || true
```

Comment:
It occurs to me that it would be better to do this in the configure script. Give the configure script a flag that lets the user specify what python they want to use. If they don't specify one, then you can guess based on whether they have `python3` or `python`, and you can complain with a good error message if they don't have any. Then the configure script makes the adjustments in the other files.

Typically, files that need adjustment use names that end with `.in`, e.g. sshdo.in. The configure script would then use `sed` to convert sshdo.in to sshdo, replacing any placeholders along the way.

Reply:
These lines have been deleted (see above).

Reply:
I've added a `--with-python=` argument to configure, and there's no patch now.